dulwich/porcelain.py

```diff
@@ -200,6 +201,11 @@ def clone(source, target=None, bare=False, checkout=None, errstream=sys.stdout,
 
     if not os.path.exists(target):
         os.mkdir(target)
+
+    # XXX This is a bit messy. Who must close this repo, and what happens if
```

Comment:
This should be fine - just wrap the code below in "try / except" and close the repo if an exception is raised (and re-raise it).